src/context/auth-context.js

```diff
@@ -0,0 +1,124 @@
+import { useContext, createContext, useState, useReducer } from "react";
+import { userInitialState, authReducer } from "reducer/auth-reducer";
+import { useToast } from "context/toast-context";
+import { LOGIN_API, SIGNUP_API } from "utils/apis";
+import axios from "axios";
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, userInitialState);
+  const { setToastVal } = useToast();
+
+  //   SignUp Functionality
+
+  const signupHandler = async (formData, setFormData, formObj) => {
+    try {
+      const response = await axios.post(SIGNUP_API, JSON.stringify(formData));
+      const { encodedToken, createdUser } = response.data;
+      localStorage.setItem("token", encodedToken);
+      authDispatch({
+        type: "SIGNUP",
+        payload: createdUser,
+        token: encodedToken,
+      });
+      setToastVal((prevVal) => ({
+        ...prevVal,
+        bg: "green",
+        msg: "Successfully Signed up",
+        isOpen: true,
+      }));
+      setTimeout(
+        () => setToastVal((prevVal) => ({ ...prevVal, isOpen: false })),
+        1500
+      );
```

Comment:
suggestion : you can keep this timeout in the toast component, just use useEffect and in the dependency have isOpen.